projects/gateway2/krtcollections/uniqueclients_test.go

```diff
@@ -139,7 +139,7 @@ func TestUniqueClients(t *testing.T) {
 				g.Eventually(func() []krtcollections.UniqlyConnectedClient {
 					allUcc = ucc.List()
 					return allUcc
-				}, "1s").Should(HaveLen(len(tc.result)))
+				}, "1s").Should(HaveLen(len(tc.result) - (i + 1)))
```

Comment:
is the kgateway check wrong then? if so can you make this fix in kgateway too?

Reply:
https://github.com/kgateway-dev/kgateway/blob/main/internal/kgateway/krtcollections/uniqueclients_test.go#L235

kgateway just checks that we emptied everything out, I think it's fine